### ISSUE TITLE
Added scripts to SabNZDB folder, and now I can't meta-tag

### ISSUE THREAD

**Issue by TheBigZig** [evilhero/mylar#2038]
I just copied my Mylar post-processing scripts to C:\Program Files\SABnzbd\scripts to get Mylar to process downloads.  Unfortunately now manual Meta-tagging doesn't seem to work.  When I try i get the following:

2018-07-29 10:52:55 | DEBUG | [MANUAL META-TAGGING] Unable to write metadata successfully - check mylar.log file.
-- | -- | --
2018-07-29 10:52:55 | WARNING | [META-TAGGER][COMIC-TAGGER][CBR-TO-CBZ]  Failed to convert cbr to cbz - check permissions on folder :  C:\mylar-master\cache/ and/or the location where Mylar is trying to tag  the files from.
2018-07-29 10:52:55 | DEBUG | [META-TAGGER]  Executing command: ['C:\\Python27\\pythonw.exe',  'C:\\mylar-master\\comictagger.py', '-e', '--delete-rar',  u'C:\\mylar-master\\cache/mylar_1lh6lz\\East of West 038.cbr']
2018-07-29 10:52:55 | DEBUG | [META-TAGGER]  Enabling ComicTagger script: ['C:\\Python27\\pythonw.exe',  'C:\\mylar-master\\comictagger.py'] with options: ['-e', '--delete-rar',  u'C:\\mylar-master\\cache/mylar_1lh6lz\\East of West 038.cbr']
2018-07-29 10:52:55 | DEBUG | [META-TAGGER] ComicTagger 1.25.1 [ninjas.walk.alone / SHURIKEN]  being used - using personal ComicVine API key supplied via mylar.
2018-07-29 10:52:55 | DEBUG | [META-TAGGER] Running the ComicTagger Add-on for Mylar
2018-07-29 10:52:55 | DEBUG | [META-TAGGER] comicpath : C:\mylar-master\cache/mylar_1lh6lz
2018-07-29 10:52:55 | DEBUG | [META-TAGGER] sabnzbdscriptpath : C:\mylar-master
2018-07-29 10:52:55 | DEBUG | [META-TAGGER] scriptname : Mylar.pyw

I've checked the permissions of the script folder, the Mylar source folder, and the folder w the CBR file in question.  Any suggestions?  
Thanks
Bill

**Comment by evilhero** [comment]
Which branch/commit of Mylar are you using? The post-processing scripts (in the post-processing folder) have nothing to do with manual metatagging - they're only used when your download client runs them after a successful download.

Was it a source installation or did you use git to install? 

If it was source, when you perform the update it will not delete the old .py files if they've been moved, so it might not be able to run properly if there's more than multiple files that perform the same function in the same directory. 

So basically, if it's a source install (in this case since you're having problems) - backup your config.ini, mylar.db and entire cache folder and then wipe the directory. Grab the latest source file, unpack it and move those files you backed up above into the new location (or same location, but new install - doesn't matter), and then try running it and seeing if that fixes the problem.

If it's a git install, then it's possibly something entirely different (and the above may not apply).

**Comment by TheBigZig** [comment]
I installed from source.  I'll try your suggestion over the weekend and let you know how it goes.  Thanks for the reply!

**Comment by TheBigZig** [comment]
I tried what you suggested, and moved my Cache folder, ini and db files.  Then I grabbed a fresh zip file from github and reinstalled.  Unfortunately, I'm still having the same issues.  Here is a grab from my logs, w debugging on:

2018-08-05 19:26:40 | DEBUG | [MANUAL META-TAGGING] Unable to write metadata successfully - check mylar.log file.
-- | -- | --
2018-08-05 19:26:40 | WARNING | [META-TAGGER][COMIC-TAGGER][CBR-TO-CBZ]  Failed to convert cbr to cbz - check permissions on folder :  C:\mylar-master\cache/ and/or the location where Mylar is trying to tag  the files from.
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] Absolute path to script: C:\Python27\python.exe
2018-08-05 19:26:40 | DEBUG | [META-TAGGER]  Executing command: ['C:\\Python27\\python.exe',  'C:\\mylar-master\\comictagger.py', '-e', '--delete-rar',  u'C:\\mylar-master\\cache/mylar_wmyhjx\\Cosmic Ghost Rider 002.cbr']
2018-08-05 19:26:40 | DEBUG | [META-TAGGER]  Enabling ComicTagger script: ['C:\\Python27\\python.exe',  'C:\\mylar-master\\comictagger.py'] with options: ['-e', '--delete-rar',  u'C:\\mylar-master\\cache/mylar_wmyhjx\\Cosmic Ghost Rider 002.cbr']
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] CBL Tagging enabled.
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] CR Tagging enabled.
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] ComicTagger 1.25.1 [ninjas.walk.alone / SHURIKEN]  being used - using personal ComicVine API key supplied via mylar.
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] Running the ComicTagger Add-on for Mylar
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] comicpath : C:\mylar-master\cache/mylar_wmyhjx
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] sabnzbdscriptpath : C:\mylar-master
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] downloadpath : C:\Users\Bill\Comics\Cosmic Ghost Rider (2018)
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] scriptname : Mylar.py
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] Paths / Locations:
2018-08-05 19:26:40 | DEBUG | New_Filepath: C:\mylar-master\cache/mylar_wmyhjx\Cosmic Ghost Rider 002.cbr
2018-08-05 19:26:40 | DEBUG | New_Folder: C:\mylar-master\cache/mylar_wmyhjx
2018-08-05 19:26:40 | DEBUG | Filename: Cosmic Ghost Rider 002.cbr
2018-08-05 19:26:40 | DEBUG | Filepath: C:\Users\Bill\Comics\Cosmic Ghost Rider (2018)\Cosmic Ghost Rider 002.cbr
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] Filename is : C:\Users\Bill\Comics\Cosmic Ghost Rider (2018)\Cosmic Ghost Rider 002.cbr
2018-08-05 19:26:40 | DEBUG | ComicTagger Path location for internal comictagger.py set to : C:\mylar-master\comictagger.py
2018-08-05 19:26:40 | DEBUG | [META-TAGGER] dirName:C:\Users\Bill\Comics\Cosmic Ghost Rider (2018)

I've checked the permissions are read AND write in all the affected folders, and still I'm getting the same error.
Any additional suggestions?
Thanks
Bill

**Comment by TheBigZig** [comment]
Tried making the changes you suggest, and Mylar still isn't tagging successfully.  Here is the log I'm getting:

2018-08-11 07:47:29 | DEBUG | Now searching experimental for issue number: 004 to try and ensure all the bases are covered
-- | -- | --
2018-08-11 07:47:26 | DEBUG | [MANUAL META-TAGGING] Unable to write metadata successfully - check mylar.log file.
2018-08-11 07:47:26 | WARNING | [META-TAGGER][COMIC-TAGGER][CBR-TO-CBZ]  Failed to convert cbr to cbz - check permissions on folder :  C:\mylar-master\cache and/or the location where Mylar is trying to tag  the files from.
2018-08-11 07:47:26 | DEBUG | [META-TAGGER] Absolute path to script: C:\Python27\pythonw.exe
2018-08-11 07:47:26 | DEBUG | [META-TAGGER]  Executing command: ['C:\\Python27\\pythonw.exe',  'C:\\mylar-master\\comictagger.py', '-e', '--delete-rar',  u'C:\\mylar-master\\cache\\mylar_fmgydo\\Uber Invasion 014.cbr']
2018-08-11 07:47:26 | DEBUG | [META-TAGGER]  Enabling ComicTagger script: ['C:\\Python27\\pythonw.exe',  'C:\\mylar-master\\comictagger.py'] with options: ['-e', '--delete-rar',  u'C:\\mylar-master\\cache\\mylar_fmgydo\\Uber Invasion 014.cbr']
2018-08-11 07:47:26 | DEBUG | [META-TAGGER] CBL Tagging enabled.
2018-08-11 07:47:26 | DEBUG | [META-TAGGER] CR Tagging enabled.
2018-08-11 07:47:26 | DEBUG | [META-TAGGER] ComicTagger 1.25.1 [ninjas.walk.alone / SHURIKEN]  being used - using personal ComicVine API key supplied via mylar.
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] Running the ComicTagger Add-on for Mylar
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] comicpath : C:\mylar-master\cache\mylar_fmgydo
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] sabnzbdscriptpath : C:\mylar-master
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] downloadpath : C:\Users\Bill\Comics\Uber Invasion (2016)
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] scriptname : Mylar.pyw
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] Paths / Locations:
2018-08-11 07:47:25 | DEBUG | New_Filepath: C:\mylar-master\cache\mylar_fmgydo\Uber Invasion 014.cbr
2018-08-11 07:47:25 | DEBUG | New_Folder: C:\mylar-master\cache\mylar_fmgydo
2018-08-11 07:47:25 | DEBUG | Filename: Uber Invasion 014.cbr
2018-08-11 07:47:25 | DEBUG | Filepath: C:\Users\Bill\Comics\Uber Invasion (2016)\Uber Invasion 014.cbr
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] Filename is : C:\Users\Bill\Comics\Uber Invasion (2016)\Uber Invasion 014.cbr
2018-08-11 07:47:25 | DEBUG | ComicTagger Path location for internal comictagger.py set to : C:\mylar-master\comictagger.py
2018-08-11 07:47:25 | DEBUG | [META-TAGGER] dirName:C:\Users\Bill\Comics\Uber Invasion (2016)

I've double checked my filesystem permissions, and made sure the the folders are read/write.  
One oddity, when I search my Mylar-master\cache folder for mylar_fmgydo, which is the folder referenced above, it doesn't show up.  Is there a way to rebuild the cache?
Also, I can send my .ini file and any logs that might help this out.
Thanks for the support!  
Bill

**Comment by evilhero** [comment]
Can you paste your config.ini file (remove the apikeys tho prior please)- as well as C:\mylar-master\lib\comictaggerlib\ct_settings\settings.ini)

Note the settings.ini file should have the full path to your winrar installation (I believe it defaults to rar_exe_path = C:\Program Files\WinRAR\Rar.exe). This is of course assuming you have winrar on your system.

**Comment by TheBigZig** [comment]
As requested, here is my config.ini file:
[General]
config_version = 8
dynamic_update = 4
launch_browser = False
auto_update = False
cache_dir = C:\mylar-master\cache
annuals_on = True
destination_dir = C:\Users\Bill\Comics
multiple_dest_dirs = None
create_folders = True
delete_remove_dir = False
nzb_startup_search = True
folder_scan_log_verbose = False
interface = default
autowant_all = False
autowant_upcoming = True
preferred_quality = 0
comic_cover_local = False
correct_metadata = False
move_files = False
rename_files = True
folder_format = $Series ($Year)
file_format = $Series $Issue
replace_spaces = False
replace_char = .
zero_level = True
zero_level_n = 00x
lowercase_filenames = False
ignore_havetotal = False
snatched_havetotal = False
syno_fix = False
search_delay = 1
grabbag_dir = C:\Users\Bill\Comics
highcount = 0
read2filename = False
send2read = False
maintainseriesfolder = False
use_minsize = False
minsize = None
use_maxsize = False
maxsize = None
add_to_csv = True
unicode_issuenumber = False
update_ended = False
upcoming_snatched = True
enable_rss = False
failed_download_handling = False
failed_auto = False
skipped2wanted = False
refresh_cache = 7
oldconfig_version = None
minimal_ini = False
wanted_tab_off = False
show_icons = False
alternate_latest_series_covers = False

[Torrents]
enable_torrents = True
minseeds = 0
enable_torrent_search = False
allow_packs = False
enable_public = False

[SABnzbd]
sab_host = http://localhost:8095/sabnzbd
sab_username = None
sab_password = None
sab_apikey = 
sab_category = Comics
sab_priority = Default
sab_to_mylar = False
sab_directory = None
sab_client_post_processing = True

[NZBGet]
nzbget_host = None
nzbget_port = None
nzbget_username = None
nzbget_password = None
nzbget_category = None
nzbget_priority = Default
nzbget_directory = None
nzbget_client_post_processing = False

[NZBsu]
nzbsu = False
nzbsu_uid = None
nzbsu_apikey = None
nzbsu_verify = True

[DOGnzb]
dognzb = True
dognzb_apikey = 
dognzb_verify = True

[Experimental]
experimental = True
altexperimental = True

[Torznab]
enable_torznab = False
extra_torznabs = 
torznab_apikey = None
torznab_category = None
torznab_host = None
torznab_verify = False
torznab_name = None

[Newznab]
newznab = False
extra_newznabs = nzbs4u, www.nzbs4u.net, 0, 47642c1db91d4a2562bbb47039e6729e, , 1

[uTorrent]
utorrent_host = None
utorrent_username = None
utorrent_password = None
utorrent_label = None

[Transmission]
transmission_host = None
transmission_username = None
transmission_password = None
transmission_directory = None

[Deluge]
deluge_host = None
deluge_username = None
deluge_password = None
deluge_label = None

[qBittorrent]
qbittorrent_host = None
qbittorrent_username = None
qbittorrent_password = None
qbittorrent_label = None
qbittorrent_folder = None
qbittorrent_startonload = False

[Prowl]
prowl_enabled = False
prowl_keys = None
prowl_onsnatch = False
prowl_priority = 0

[NMA]
nma_enabled = False
nma_apikey = None
nma_priority = 0
nma_onsnatch = False

[PUSHOVER]
pushover_enabled = False
pushover_apikey = None
pushover_userkey = None
pushover_priority = 0
pushover_onsnatch = False
pushover_device = None

[BOXCAR]
boxcar_enabled = False
boxcar_onsnatch = False
boxcar_token = None

[PUSHBULLET]
pushbullet_enabled = True
pushbullet_apikey = 
pushbullet_deviceid = None
pushbullet_channel_tag = None
pushbullet_onsnatch = False

[TELEGRAM]
telegram_enabled = False
telegram_token = None
telegram_userid = None
telegram_onsnatch = False

[SLACK]
slack_enabled = False
slack_webhook_url = None
slack_onsnatch = False

[Client]
nzb_downloader = 0
torrent_downloader = 4

[PostProcess]
pre_scripts = None
snatch_script = None
file_opts = move
extra_scripts = None
enable_snatch_script = False
enable_check_folder = False
local_torrent_pp = False
check_folder = None
snatchedtorrent_notify = False
enable_extra_scripts = False
post_processing = True
post_processing_script = None
enable_pre_scripts = False

[Metatagging]
setdefaultvolume = False
cbr2cbz_only = True
unrar_cmd = None
ct_cbz_overwrite = False
enable_meta = True
ct_settingspath = None
cmtag_volume = True
cmtagger_path = None
cmtag_start_year_as_volume = False
ct_tag_cbl = True
ct_tag_cr = True

[Weekly]
weekfolder = False
weekfolder_format = 0
alt_pull = 0
weekfolder_loc = None
biggie_pub = 55
pull_refresh = None
indie_pub = 75
pack_0day_watchlist_only = True

[Scheduler]
search_interval = 360
rss_checkinterval = 20
download_scan_interval = 5
check_github_interval = 360

[Interface]
http_password = None
https_chain = None
http_port = 8090
http_root = /
https_force_on = False
http_username = None
https_cert = None
http_host = 0.0.0.0
enable_https = False
https_key = None
host_return = None
authentication = 0
login_timeout = 43800

[Import]
imp_metadata = False
comic_dir = C:\Users\Bill\Comics
imp_move = False
imp_rename = False
add_comics = True

[Perms]
chmod_file = 0660
chowner = None
enforce_perms = True
chmod_dir = 0777
chgroup = None

[CV]
cv_verify = False
cv_only = True
cvinfo = False
blacklisted_publishers = None
comicvine_api = 074c8c0ab16a3437da83d60229775508b9cac631
cv_onetimer = True
cvapi_rate = 2

[Rtorrent]
rtorrent_ca_bundle = None
rtorrent_directory = None
rtorrent_host = None
rtorrent_label = None
rtorrent_startonload = False
rtorrent_verify = False
rtorrent_password = None
rtorrent_authentication = basic
rtorrent_rpc_url = None
rtorrent_ssl = False
rtorrent_username = None

[32P]
mode_32p = False
password_32p = None
verify_32p = True
search_32p = False
username_32p = None
deep_search_32p = False
passkey_32p = None
rssfeed_32p = None
enable_32p = False

[Git]
git_user = evilhero
check_github = True
git_path = None
check_github_on_startup = True
git_branch = master

[AutoSnatch]
auto_snatch_script = None
pp_sshhost = None
auto_snatch = False
pp_sshuser = None
pp_sshlocalcd = None
pp_sshport = 22
pp_sshkeyfile = None
pp_sshpasswd = None

[TPSE]
tpse_verify = True
tpse_proxy = None
enable_tpse = False

[Providers]
usenet_retention = 1500
provider_order = 0, dognzb, 1, Experimental

[StoryArc]
storyarcdir = False
arc_folderformat = $arc ($spanyears)
copy2arcdir = False
arc_fileops = copy
search_storyarcs = False
upcoming_storyarcs = False

[Duplicates]
ddump = False
duplicate_dump = None
dupeconstraint = filesize

[Blackhole]
blackhole_dir = None

[Update]
newcom_dir = None
fftonewcom_dir = False
locmove = False

[Logs]
log_dir = C:\mylar-master\logs
max_logsize = 1000000
log_level = 0
max_logfiles = 5

[Watchdir]
torrent_local = False
local_watchdir = C:\Users\Bill\My Torrents

[Tablet]
tab_host = None
tab_directory = None
tab_user = None
tab_enable = False
tab_pass = None

[TEST]
test_value = True

[OPDS]
opds_metainfo = False
opds_authentication = False
opds_username = None
opds_password = None
opds_enable = False

[Seedbox]
seedbox_port = None
seedbox_user = None
torrent_seedbox = False
seedbox_pass = None
seedbox_watchdir = None
seedbox_host = None

[API]
api_key = None
api_enabled = True

And my C:\mylar-master\lib\comictaggerlib\ct_settings\settings.ini file.  I've been using 7-zip

[settings]
check_for_new_version = True
rar_exe_path = ""
unrar_exe_path = ""
send_usage_stats = False

I see the RAR path is blank, and I'm sure that's a problem.  The odd thing is that before I started this thread it was working fine.  So did something change that path?  Also, I've been using 7-zip for my RAR purposes.  Is that an issue?
Thanks, and really, I mean it.  I can't imaging  you are making $ off this, so the level of support is amazing.
Bill

**Comment by evilhero** [comment]
I don't think 7-zip will work as comictagger makes specific calls to the unrar executable with switches which I'd be surprised if 7-zip would work with.

Can you try installing WinRAR and then point the unrar_exe_path in the settings.ini file to the full path?

Also, you might want to change these values in your config.ini :
alt_pull = 2 (far more efficient pullist usage) 
cv_verify = True
cbr2cbz_only = False (this will only perform a cbr to cbz conversion and will not apply any metatagging if set to True)

If you enable debug logging prior to performing the metatagging, and if it doesn't work you can paste the mylar.log file showing the metatagging as there might be additional info in the actual log file as opposed to the log GUI (specifically when the conversion fails it should log more specific information related to the error)
